tests/core/abi-utils/test_abi_utils.py

```diff
@@ -104,58 +485,583 @@ def test_fn_abi_to_4byte_selector(fn_abi, expected):
+        ),
+        (
+            ABI_FUNCTION_MULTI_TYPED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "bytes32"}],
+        ),
+        (
+            ABI_CONSTRUCTOR_WITH_INPUT,
+            [{"name": "inputs", "type": "str"}],
+        ),
+        (
+            ABI_ERROR_INVALID,
+            [{"name": "a", "type": "address"}, {"name": "b", "type": "bytes32"}],
+        ),
     ],
-}
+)
+def test_get_input_types_from_abi_element(
+    abi_element: ABIElement, input_args: Sequence[ABIElement]
+) -> None:
+    assert get_abi_input_types(abi_element) == [arg.get("type") for arg in input_args]
+
+
+@pytest.mark.parametrize(
+    "abi_element",
+    (
+        ABI_FALLBACK,
+        ABI_RECEIVE,
+    ),
+)
+def test_get_input_types_raises_for_fallback_or_receive_abi(
+    abi_element: ABIElement,
+) -> None:
+    with pytest.raises(
+        ValueError,
+        match=f"Inputs not supported for function types 'fallback' or 'receive'."
+        f" Provided ABI type was '{abi_element['type']}'.",
+    ):
+        get_abi_input_types(abi_element)
+
+
+@pytest.mark.parametrize(
+    "abi_element,outputs",
+    [
+        (
+            ABI_FUNCTION_SINGLE_ARG,
+            [{"name": "arg0", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_TWO_NAMED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_MULTI_TYPED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "bytes32"}],
+        ),
+        (
+            ABI_FUNCTION_TWO_UNNAMED_ARGS,
+            [],
+        ),
+    ],
+)
+def test_get_abi_output_names(
+    abi_element: ABIElement, outputs: Sequence[ABIElement]
+) -> None:
+    assert get_abi_output_names(abi_element) == [
+        output.get("name") for output in outputs
+    ]
+
+
+@pytest.mark.parametrize(
+    "abi_element",
+    (
+        (ABI_FALLBACK),
+        (ABI_RECEIVE),
+        (ABI_CONSTRUCTOR),
+        (ABI_EVENT_LOG_NO_ARG),
+        (ABI_ERROR),
+    ),
+)
+def test_get_abi_output_names_raises_for_non_function_types(
+    abi_element: ABIElement,
+) -> None:
+    with pytest.raises(
+        ValueError,
+        match=f"Outputs only supported for ABI type 'function'."
+        f" Provided ABI type was '{abi_element['type']}'.",
+    ):
+        get_abi_output_names(abi_element)
+
+
+@pytest.mark.parametrize(
+    "abi_element,outputs",
+    [
+        (
+            ABI_FUNCTION_SINGLE_ARG,
+            [{"name": "arg0", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_TWO_NAMED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_MULTI_TYPED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "bytes32"}],
+        ),
+    ],
+)
+def test_get_abi_output_types(
```

Comment:
I think this test suffers from truly testing the functionality based on my above comment. All of the `inputs` in these ABI examples are identical to the `outputs`. I think we should use different inputs and outputs to ensure, without question for a reader, that these tests are doing what we think they're doing.

Reply:
I went through these and added tests consistently across each of them. I think the varying input and output types are covered now but can add more if you spot any other cases!

Reply:
I think `test_abi_get_input_types_from_abi_element()` ended up missing [some of the new test cases](https://github.com/ethereum/eth-utils/pull/271/files#diff-dd3235d6149da0f1637a95fda56a179ddfb1125a64939c032fabcffacd12e8c6R799-R818) you added to `test_abi_get_input_names_from_abi_element()`.

Reply:
We should try to make sure that we're testing some of the same things across them for consistency that all `name` extraction tests are using the same exampels as the `type` extraction tests.